packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -268,6 +268,9 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
 
   @override
   Future<FlutterCommandResult> verifyThenRunCommand(String? commandPath) {
+    if (stringArg('device-id', global: true) != null) {
+      throwToolExit('The --device-id (-d) option is not supported with the "test" sub-command.');
```

Comment:
Nit: should we ask the user if they meant to use --platform instead?
```suggestion
      throwToolExit('The --device-id (-d) option is not supported with the "test" sub-command. Did you mean to use --platform instead?');
```

Reply:
Given that we've marked `--platform` as deprecated, I don't think we should nudge users toward it.